### PR TITLE
Logic fix for changed external behavour

### PR DESCRIPTION
**Pull request by zigszigsdk**
As discussed in https://github.com/ffMathy/FluffySpoon.AspNet.LetsEncrypt/pull/20, letsencrypt must've changed their API responses, so we need to change this project accordingly.

### CODE REVIEW

File: src/FluffySpoon.AspNet.LetsEncrypt/LetsEncryptRenewalService.cs [hunk 1]
```diff
@@ -273,31 +272,37 @@ private async Task ValidateOrderAsync(string[] domains, IOrderContext order)
...
 
-				if (challengeExceptions.Length > 0)
-					throw new OrderInvalidException(
+				if (challengeExceptions.Length > 0) 
+                {
+					var exception = new OrderInvalidException(
 						"One or more LetsEncrypt orders were invalid. Make sure that LetsEncrypt can contact the domain you are trying to request an SSL certificate for, in order to verify it.",
 						new AggregateException(challengeExceptions));
+
+                    _logger.LogTrace("Throwing exception: ", exception);
```
Comment (ffMathy) [inline src/FluffySpoon.AspNet.LetsEncrypt/LetsEncryptRenewalService.cs]: Can we get rid of this? Shouldn't the exception in itself be useful?

File: src/FluffySpoon.AspNet.LetsEncrypt/LetsEncryptRenewalService.cs [hunk 2]
```diff
@@ -273,31 +272,37 @@ private async Task ValidateOrderAsync(string[] domains, IOrderContext order)
...
 
 			while (true)
 			{
-				if (!challenges.Any(x => x.Status == ChallengeStatus.Pending || x.Status == ChallengeStatus.Processing))
+				if (challenges.Any(x => x.Status == ChallengeStatus.Valid) || challenges.All(x => x.Status == ChallengeStatus.Invalid))
 					break;
-
-				await Task.Delay(1000);
+                
+                _logger.LogTrace("Still pending or processing..");
```
Comment (ffMathy) [inline src/FluffySpoon.AspNet.LetsEncrypt/LetsEncryptRenewalService.cs]: Can we remove this log statement?

Comment (zigszigsdk) [reply src/FluffySpoon.AspNet.LetsEncrypt/LetsEncryptRenewalService.cs]: Done

File: src/FluffySpoon.AspNet.LetsEncrypt/LetsEncryptRenewalService.cs (hunk 1 above)
Comment (zigszigsdk) [reply src/FluffySpoon.AspNet.LetsEncrypt/LetsEncryptRenewalService.cs]: Done

**Comment by zigszigsdk** [comment]
Mange tak, og lige over! ;)

**Pull request merged**